autogalaxy/analysis/analysis.py

```diff
@@ -1,36 +1,92 @@
 from astropy import cosmology as cosmo
 import json
 import numpy as np
+from typing import Optional, Union
 from os import path
 
 from autoconf import conf
 import autofit as af
 import autoarray as aa
 
 from autogalaxy import exc
+from autogalaxy.hyper.hyper_data import HyperImageSky
+from autogalaxy.hyper.hyper_data import HyperBackgroundNoise
 from autogalaxy.galaxy.galaxy import Galaxy
 from autogalaxy.plane.plane import Plane
+from autogalaxy.analysis.result import ResultDataset
 
 
 class Analysis(af.Analysis):
-    def __init__(self, hyper_dataset_result=None, cosmology=cosmo.Planck15):
-        self.hyper_dataset_result = hyper_dataset_result
+    def __init__(self, cosmology=cosmo.Planck15):
+        """
+        Analysis classes are used by PyAutoFit to fit a model to a dataset via a non-linear search.
+
+        This abstract Analysis class for all model-fits which fit galaxies (or objects containing galaxies like a
+        plane), but does not perform a model-fit by itself (and is therefore only inherited from).
+
+        This class stores the Cosmology used for the analysis and hyper datasets used for certain model classes.
+
+        Parameters
+        ----------
+        cosmology
+            The AstroPy Cosmology assumed for this analysis.
+        """
         self.cosmology = cosmology
 
+    def plane_for_instance(self, instance: af.ModelInstance) -> Plane:

```

Comment:
lol this method might not need to exist.

If it gets overridden later then consider using a classmethod.